### PR TITLE
Update MPASSIT

### PR DESCRIPTION
**Pull request by EricJames-NOAA**
This PR updates to the latest version of MPASSIT from github.  This gets rid of the x/y streak issue for the u/v variables, and also allows us to remove dependency on geo_em.d01.nc file.

### CODE REVIEW

File: scripts/exrrfs_mpassit.sh [hunk 1]
```diff
@@ -37,7 +37,8 @@ module list
 set -x  
 ### temporarily solution since mpassit uses different modules files that other components
 source prep_step
-srun /lfs5/BMC/nrtrr/FIX_RRFS2/exec/mpassit.x namelist.mpassit  #gge.debug temp solution
+#srun /lfs5/BMC/nrtrr/FIX_RRFS2/exec/mpassit.x namelist.mpassit  #gge.debug temp solution
+srun /lfs4/BMC/wrfruc/ejames/MPASSIT/bin/mpassit namelist.mpassit
```
Comment (guoqing-noaa) [inline scripts/exrrfs_mpassit.sh:41]: can you remove line 40, and replace line 41 with this?
```
srun /lfs5/BMC/nrtrr/FIX_RRFS2/exec/mpassit_20240801.x namelist.mpassit
```
If we can build mpassit from source code and make it work on both Hera and Jet. That will be sweeter.

File: parm/rrfs/namelist.mpassit [hunk 2]
```diff
@@ -2,10 +2,19 @@
...
     output_file          = "mpassit.@timestr@.nc"
+    target_grid_type     = 'lambert'
     interp_diag          = .true.
     interp_hist          = .true.
     wrf_mod_vars         = .true.
     esmf_log             = .false.
+    nx = 480
+    ny = 280
+    dx = 12000.0
+    dy = 12000.0
```
Comment (guoqing-noaa) [inline parm/rrfs/namelist.mpassit:14]: We will assign dx and dy on the fly in `exrrfs_mpassit.sh` based on "NET" (conus12km or conus3km).

You may refer to this file: `exrrfs_fcst.sh` on using `sed` to generate a final namelist file.
https://github.com/rrfs2/rrfs-workflow/blob/1ebd8a2025a86be0ce06b5c6b37e6906e148201f/scripts/exrrfs_fcst.sh#L55

**Review by guoqing-noaa — COMMENTED** [review]
This is great! Thank you, @EricJames-NOAA 
I have two comments.

File: parm/rrfs/namelist.mpassit (hunk 2 above)
Comment (guoqing-noaa) [reply parm/rrfs/namelist.mpassit:14]: I think nx,ny will change for different `NET` as well.

File: scripts/exrrfs_mpassit.sh (hunk 1 above)
Comment (EricJames-NOAA) [reply scripts/exrrfs_mpassit.sh:41]: @guoqing-noaa thanks for the suggestion.  Do we want to try to grab a specific hash from github and then compile it?  I guess we can try to follow what we do for MPAS, WPS, and WRF?

File: parm/rrfs/namelist.mpassit (hunk 2 above)
Comment (EricJames-NOAA) [reply parm/rrfs/namelist.mpassit:14]: @guoqing-noaa thanks I'll see if I can get that working!

File: scripts/exrrfs_mpassit.sh (hunk 1 above)
Comment (guoqing-noaa) [reply scripts/exrrfs_mpassit.sh:41]: Thanks! Following MPAS and WPS is a good idea.

As the first step, if you can compile MPASSIT from a fresh clone on both Hera/Jet, that will be good.

Adding the mpassit source code into the workflow will involve the git submodule commands. We can work together to add mpassit as a submodule.

Comment (EricJames-NOAA) [reply scripts/exrrfs_mpassit.sh:41]: Sounds good!  For now, I will compile MPASSIT from a fresh clone on Hera.

File: parm/rrfs/namelist.mpassit [hunk 3]
```diff
@@ -8,13 +8,13 @@
...
     esmf_log             = .false.
-    nx = 480
-    ny = 280
-    dx = 12000.0
-    dy = 12000.0
-    ref_lat = 39.0
+    nx = @nx@
+    ny = @ny@
+    dx = @dx@.0
+    dy = @dx@.0
```
Comment (guoqing-noaa) [inline parm/rrfs/namelist.mpassit:14]: Suggest using `dx=@dx@` and `dy=@dx@`

File: parm/rrfs/namelist.mpassit [hunk 4]
```diff
@@ -8,13 +8,13 @@
...
+    ref_lat = @lat@.@lat2@
     ref_lon = -97.5
-    truelat1 = 39.0
-    truelat2 = 39.0
+    truelat1 = @lat@.@lat2@
+    truelat2 = @lat@.@lat2@
```
Comment (guoqing-noaa) [inline parm/rrfs/namelist.mpassit:18]: suggest using `ref_lat = @ref_lat@`, `truelat1 = @ref_lat@` and `truelat2 = @ref_lat@`

**Review by guoqing-noaa — APPROVED** [review]
Looks good to me. Hope we can get MPASSIT work on Hera soon. Thanks for your work on this!

**Pull request merged**